### PR TITLE
Issue #38: Created the splash screen layout and MVVM structure

### PR DESCRIPTION
**Pull request by Amanda-Kolopa**
#38 progress:

- Created a splash screen that will display the application's logo and name when launched.
- Set up the code to follow MVVM since the splash screen is used for performing the GoogleMapsAPI instantiations before calling MapFragment
(right now a thread delay of 3 seconds is simulating the connection time).
- Remaining work: refactor the instantiation code in MapFragment into a MapRepository to be initiated through SplashViewModel

### CODE REVIEW

**Comment by Neeham** [comment]
Great work @Amanda-Kolopa It looks amazing 😊 

On my end, when the app opens both the left and top of the image has a small thin white border. Could you please fix it. Have a look at the image:

Screenshot_20200224-154421_Campus Compass

Thank you! 😄

File: app/src/main/java/com/droidhats/campuscompass/viewmodels/SplashViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import androidx.lifecycle.ViewModel
+
+class SplashViewModel : ViewModel() {
+
+    //set as false to identify instantiations are not complete
+    private val mutableLiveData = MutableLiveData<Boolean>().apply {
+        value = false
+    }
+
+    //example of a value that can be accessed from SplashFragment
```
Comment (NicholasNagy) [inline app/src/main/java/com/droidhats/campuscompass/viewmodels/SplashViewModel.kt]: What do you mean by example? Is this code necessary or not?

File: app/src/main/java/com/droidhats/campuscompass/viewmodels/SplashViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    val hardCode: LiveData<Boolean> = mutableLiveData
+
+    fun init() {
+        mutableLiveData.value = true //hard coded to show successful API connection
+        returnData() //dummy function - represents Map initialization methods
```
Comment (NicholasNagy) [inline app/src/main/java/com/droidhats/campuscompass/viewmodels/SplashViewModel.kt]: I don't think we need a dummy function here, especially if it doesn't serve any purpose

File: app/src/main/java/com/droidhats/campuscompass/views/MapFragment.kt [hunk 3]
```diff
@@ -82,7 +83,13 @@ class MapFragment : Fragment(), OnMapReadyCallback, GoogleMap.OnMarkerClickListe
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
-        return inflater.inflate(R.layout.map_fragment, container, false)
+        val mapFragment = inflater.inflate(R.layout.map_fragment, container, false)
+        requireActivity().onBackPressedDispatcher.addCallback(object : OnBackPressedCallback(true) {
+            override fun handleOnBackPressed() {
+                //do nothing on purpose - user can't go back to splash screen
```
Comment (NicholasNagy) [inline app/src/main/java/com/droidhats/campuscompass/views/MapFragment.kt]: we should be exiting the application here actually. We don't want them to be stuck inside the application...

File: app/src/main/java/com/droidhats/campuscompass/views/MapFragment.kt [hunk 4]
```diff
@@ -288,8 +295,8 @@ class MapFragment : Fragment(), OnMapReadyCallback, GoogleMap.OnMarkerClickListe
 
             //Calculating the center of the polygon to use for it's location.
             // This won't be necessary once we hold the Buildings in a common class
-            var centerLat: Double = 0.0
-            var centerLong: Double = 0.0
```
Comment (NicholasNagy) [inline app/src/main/java/com/droidhats/campuscompass/views/MapFragment.kt]: Please do not remove types to our variables, it'll make it harder to debug if this becomes a habit

Comment (Amanda-Kolopa) [reply app/src/main/java/com/droidhats/campuscompass/views/MapFragment.kt]: Sorry!

File: app/src/main/java/com/droidhats/campuscompass/viewmodels/SplashViewModel.kt (hunk 1 above)
Comment (Amanda-Kolopa) [reply app/src/main/java/com/droidhats/campuscompass/viewmodels/SplashViewModel.kt]: It's not necessary, it was originally there to show how data would be passed between fragments but I forgot to update it. Good catch!

**Comment by makchamp** [comment]
After fixing some minor things(check commits for details) everything appears to be working fine. It looks quite good !

**Pull request merged**